demo/style_configurator.c

```diff
@@ -0,0 +1,2822 @@
+// manually set those after calling the function by accessing ctx->style->*?
+
+// style_general? pass array in instead of static?
+static void
+style_colors(struct nk_context* ctx, struct nk_color color_table[NK_COLOR_COUNT])
+{
+	const char* color_labels[NK_COLOR_COUNT] =
+	{
+		"COLOR_TEXT:",
+		"COLOR_WINDOW:",
+		"COLOR_HEADER:",
+		"COLOR_BORDER:",
+		"COLOR_BUTTON:",
+		"COLOR_BUTTON_HOVER:",
+		"COLOR_BUTTON_ACTIVE:",
+		"COLOR_TOGGLE:",
+		"COLOR_TOGGLE_HOVER:",
+		"COLOR_TOGGLE_CURSOR:",
+		"COLOR_SELECT:",
+		"COLOR_SELECT_ACTIVE:",
+		"COLOR_SLIDER:",
+		"COLOR_SLIDER_CURSOR:",
+		"COLOR_SLIDER_CURSOR_HOVER:",
+		"COLOR_SLIDER_CURSOR_ACTIVE:",
+		"COLOR_PROPERTY:",
+		"COLOR_EDIT:",
+		"COLOR_EDIT_CURSOR:",
+		"COLOR_COMBO:",
+		"COLOR_CHART:",
+		"COLOR_CHART_COLOR:",
+		"COLOR_CHART_COLOR_HIGHLIGHT:",
+		"COLOR_SCROLLBAR:",
+		"COLOR_SCROLLBAR_CURSOR:",
+		"COLOR_SCROLLBAR_CURSOR_HOVER:",
+		"COLOR_SCROLLBAR_CURSOR_ACTIVE:",
+		"COLOR_TAB_HEADER:"
+	};
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+	for (int i=0; i<NK_COLOR_COUNT; ++i) {
+		nk_label(ctx, color_labels[i], NK_TEXT_LEFT);
+
+		if (nk_combo_begin_color(ctx, color_table[i], nk_vec2(nk_widget_width(ctx), 400))) {
+			nk_layout_row_dynamic(ctx, 120, 1);
+			struct nk_colorf colorf = nk_color_picker(ctx, nk_color_cf(color_table[i]), NK_RGB);
+			nk_layout_row_dynamic(ctx, 25, 1);
+			colorf.r = nk_propertyf(ctx, "#R:", 0, colorf.r, 1.0f, 0.01f,0.005f);
+			colorf.g = nk_propertyf(ctx, "#G:", 0, colorf.g, 1.0f, 0.01f,0.005f);
+			colorf.b = nk_propertyf(ctx, "#B:", 0, colorf.b, 1.0f, 0.01f,0.005f);
+
+			color_table[i] = nk_rgb_cf(colorf);
+
+			nk_combo_end(ctx);
+		}
+	}
+
+	nk_style_from_table(ctx, color_table);
+}
+
+static void
+style_text(struct nk_context* ctx, struct nk_style_text* out_style)
+{
+	struct nk_style_text text = *out_style;
+	//text = &style->text;
+	//text->color = table[NK_COLOR_TEXT];
+	//text->padding = nk_vec2(0,0);
+
+	char buffer[64];
+
+	nk_layout_row_dynamic(ctx, 30, 2);
+	nk_label(ctx, "Color:", NK_TEXT_LEFT);
+	if (nk_combo_begin_color(ctx, text.color, nk_vec2(nk_widget_width(ctx), 400))) {
+		nk_layout_row_dynamic(ctx, 120, 1);
+		struct nk_colorf colorf = nk_color_picker(ctx, nk_color_cf(text.color), NK_RGB);
+		nk_layout_row_dynamic(ctx, 25, 1);
+		colorf.r = nk_propertyf(ctx, "#R:", 0, colorf.r, 1.0f, 0.01f,0.005f);
+		colorf.g = nk_propertyf(ctx, "#G:", 0, colorf.g, 1.0f, 0.01f,0.005f);
+		colorf.b = nk_propertyf(ctx, "#B:", 0, colorf.b, 1.0f, 0.01f,0.005f);
+
+		text.color = nk_rgb_cf(colorf);
+
+		nk_combo_end(ctx);
+	}
+
+	nk_label(ctx, "Padding:", NK_TEXT_LEFT);
+	sprintf(buffer, "%.2f, %.2f", text.padding.x, text.padding.y);
+	if (nk_combo_begin_label(ctx, buffer, nk_vec2(200,200))) {
+		nk_layout_row_dynamic(ctx, 25, 1);
+		nk_property_float(ctx, "#X:", -100.0f, &text.padding.x, 100.0f, 1,0.5f);
+		nk_property_float(ctx, "#Y:", -100.0f, &text.padding.y, 100.0f, 1,0.5f);
+		nk_combo_end(ctx);
```

Comment:
Same here. `style_pointer_sliders` would be a good function.

Reply:
Yeah, like I said, this was definitely a first draft and I did a lot of copy-pasting and was mostly focused on just getting it to work and exploring the very large style hierarchy.  I tend to follow something akin to Casey Muratori's ["Semantic Compression"](https://caseymuratori.com/blog_0015) and posted this without any additional "passes".  I'm not sure about hitting 1000 lines though, we'll see.  I kind of want to keep the comments in there even though they're just pasted from nuklear.h; they're a quick reference for all the elements each function can change and those alone are a couple hundred lines.  I'll try to finish up these changes today or tomorrow.

Reply:
Sounds good. Do what you think needs done 👍